hera_mc/cm_hookup.py

```diff
@@ -415,78 +89,90 @@ def get_hookup(self, hpn_list, rev='ACTIVE', port_query='all', at_date='now', ex
-        return hookup_dict
+        return self.get_hookup_from_db(hpn=hpn, pol=pol, at_date=at_date,
+                                       exact_match=exact_match, hookup_type=hookup_type)
 
-    def _requested_list_OK_for_cache(self, hpn_list):
+    def requested_list_OK_for_cache(self, hpn):
         """
         Checks that all hookup requests match the cached keys.
 
         Parameters
         ----------
-        hpn_list : list
+        hpn : list
             List of HERA part numbers being checked.
 
         Returns
         -------
         bool
             True if the part numbers are in the cached keys. Else False
         """
-        for x in hpn_list:
+        for x in hpn:
             for key_prefix in self.hookup_list_to_cache:
                 if x[:len(key_prefix)].upper() == key_prefix.upper():
                     break
             else:
                 return False
         return True
 
-    def get_hookup_from_db(self, hpn_list, rev, port_query, at_date, exact_match=False, hookup_type=None):
+    def _cull_dict(self, hpn, search_dict, exact_match):
+        """
+        Determines the complete appropriate set of parts to use within search_dict as
+        specified by hpn and exact_match.  It is not case sensitive.
+
+        Parameters
+        ----------
+        hpn : list
+            Contains part numbers (or partial part numbers) to include in the hookup list.
+        search_dict : dict
+            Contains information about all parts possible to search, keyed on the "standard"
+            cm_utils.make_part_key
+        exact_match : bool
+            If False, will only check the first characters in each hpn entry.  E.g. 'HH1'
+            would allow 'HH1', 'HH10', 'HH123', etc
+
+        Returns
+        -------
+        dict
+            Contains the found entries within search_dict
+        """
+
+        hpn_upper = [x.upper() for x in hpn]
+        found_dict = {}
+        for key in search_dict.keys():
+            hpn, rev = cm_utils.split_part_key(key.upper())
+            use_this_one = False
+            if exact_match:
+                if hpn in hpn_upper:
+                    use_this_one = True
+            else:
+                for hlu in hpn_upper:
+                    if hpn.startswith(hlu):
+                        use_this_one = True
+                        break
+            if use_this_one:
+                found_dict[key] = copy.copy(search_dict[key])
+        return(found_dict)
+
+    def get_hookup_from_db(self, hpn, pol, at_date, exact_match=False, hookup_type=None):
         """
         This gets called by the get_hookup wrapper if the database needs to be read (for instance, to generate
-        a cache file, or search for parts different than those keyed on in the cache file.)
+        a cache file, or search for parts different than those keyed on in the cache file.)  It will look over
+        all active revisions.
```

Comment:
Again, one simple line in declarative voice, then an empty line, then a fuller description if needed.

Reply:
Done.

Reply:
Really? I see three lines of continuous text.

Reply:
I don't -- only one was more than one line, and I just fixed that.